controllers/api/userRoutes.js

```diff
@@ -0,0 +1,108 @@
+const router = require('express').Router();
+const { User, GuestList, Food} = require('../../models');
+
+
+// getting all the users with the guestlist
+router.get('/', (req, res) => {
+    // find all users including its associated guestLists
+    User.findAll({
+        include: [
+            {
+                model: GuestList,
+                attributes: ['id', 'first_name', 'last_name', 'street', 'city', 'state', 'zipcode', 'phone_number', 'email', 'rsvp', 'user_id', 'food_id' ]
+            }
+          ]
+      })
+    .then(dbUserData => res.json(dbUserData))
+    .catch(err => {
+        console.log(err);
+        res.status(500).json(err);
+    });
+});
+
+
+router.get('/:id', (req, res) => {
+    // find one user by its `id` value including its associated guestLists
+    User.findOne({
+        where: {
+            id: req.params.id
+        },
+        include: [
+            {
+                model: GuestList,
+                attributes: ['id', 'first_name', 'last_name', 'street', 'city', 'state', 'zipcode', 'phone_number', 'email', 'rsvp', 'user_id', 'food_id' ]
+            }
+
+        ]
+      })
+      .then(dbUserData => dbUserData ? res.json(dbUserData) : res.json({message: 'User not found'}))
+      .catch(err => {
+          console.log(err);
+          res.status(500).json(err);
+     });
+  
+});
+
+
+router.post('/', (req, res) => {
+    // create a new user
+    User.create({
+        user_first_name: req.body.first_name,
+        user_last_name: req.body.last_name,
+        user_email: req.body.email,
+        user_password: req.body.password
+    })
+    .then(dbUserData => res.json(dbUserData))
+    .catch(err => {
+          console.log(err);
+          res.status(500).json(err);
+     });
+  
+});
+
+//?????
+// trying to figure out if we want to update guestList 
+// associated with particular guest
+//  or update only user info
+
+router.put('/:id', (req, res) => {
+    // update product data
+    User.update(req.body, {
+      where: {
+        id: req.params.id,
+      },
+    })
+      .then((user) => {
+        // find all associated guestList 
+        console.log(user);
+        return GuestList.findAll({ where: { user_id: req.params.id } });
+      })
+      .then((guestList) => {
+        //  stuck here
```

Comment:
We only want to update the user info.  so it would look like this `router.put("/:id", (req, res) => {
  User.update(req.body, {
    where: {
      id: req.params.id,
    },
  })
    .then((dbUserData) => {
      if (!dbUserData) {
        res.status(404).json({ message: "No user found with this id" });
        return;
      }
      res.json(dbUserData);
    })
    .catch((error) => {
      console.log(error);
      res.status(500).json(error);
    });
});`